src/3d/Desugar.fst

```diff
@@ -180,21 +195,53 @@ let resolve_ident (env:qenv) (i:ident) : ML ident =
                         i.range
              else i)  //return the local name as is
        else (match i.v.modul_name with  //it's a top-level name
-             | None -> { i with v = { i.v with modul_name = Some env.mname } }  //if unqualified, add current module
+             | None -> maybe_resolve_as_ifdef i
              | Some m ->  //if already qualified, check if it is an abbreviation
                (match H.try_find env.module_abbrevs m with
                 | None -> i
                 | Some m -> { i with v = { i.v with modul_name = Some m } }))
 
 
-let rec resolve_expr' (env:qenv) (e:expr') : ML expr' =
+let rec collect_ifdef_guards (env:qenv) (e:expr) 
+  : ML unit
+  = let check_resolved_to_ifdef i =
+      match env.global_env.ge_cfg with
+      | None -> false
+      | Some (cfg, cfg_module_name) ->
+        List.mem i.v.name cfg.compile_time_flags.flags
+        && i.v.modul_name = Some cfg_module_name
+    in
+    match e.v with
+    | This -> error "'this' is not allowed in the guard of an #if" e.range
+    | Static _ -> failwith "static should have been eliminated already"
+    | Constant _ -> ()
+    | Identifier i ->

```

Comment:
ensures that all variables in a #if are declared as compile-time flags in the config